### ISSUE TITLE
[BUG] LinearRegressionModel can't predict, saying input X contains NaN despite having no missing value

### ISSUE THREAD

**Issue by torivor** [unit8co/darts#1895]
**Describe the bug**
I am using a LinearRegressionModel to forecast a hierarchical, hourly demand dataset. The demand is intermittent as it has 2 kind of time periods: a period which always have zero demand and a period which always have more than 1 demand.

Therefore, I ensured that my dataset has time series index with hourly frequency before converting it into a darts time series. I filled all the missing values with 0.0 (float representation of 0) and thus there shouldn't be any missing value. The missing value ratio function provided within the darts library have also confirmed this:
![image](https://github.com/unit8co/darts/assets/59689052/b5fa7bfb-4459-4374-bee2-8c288b4eefee)

Thus, I proceed with using the data for modelling and prediction:
![image](https://github.com/unit8co/darts/assets/59689052/83f7c9d6-4721-4f4c-bf9f-3eebbc4419b4)

However, during prediction, I encountered an error saying that input X contains NaN:
![image](https://github.com/unit8co/darts/assets/59689052/a62bcad7-b5e0-49fe-b6ca-41b8b32b245a)
...
![image](https://github.com/unit8co/darts/assets/59689052/e6137958-4dc9-479c-a0af-801c1a4e3888)

**Expected behavior**
This shouldn't have had happened right? as I have already checked before that the input doesn't contain any missing value. Is there any solution to this? Thank you

**System (please complete the following information):**
 - Python version: 3.9.16
 - darts version: 0.24.0

**Additional context**
I cant share the dataset since it is strictly confidential. However, the dataset characteristic can be easily understood as the dataset is basically a hourly record of the amount of customers served by a business unit.

**Comment by madtoinou** [comment]
Hi @torivor,

Can you share the code snippet where you create the `TimeSeries`, the output of `missing_values_ratio()` as well as the part where you replace the NaN with zeros?

You can check the presence of NaN in the `TimeSeries` by using `np.isnan(ts.all_values()).sum()`, I would trust it more than the method mentioned above.

**Comment by torivor** [comment]
Sure @madtoinou, here are the snippets you have requested:

Code snippet where I have created the TimeSeries:
![image](https://github.com/unit8co/darts/assets/59689052/5fa7424a-19f0-400e-bacd-5dd7d02fd2a3)

The output of missing_values_ratio():
![image](https://github.com/unit8co/darts/assets/59689052/e8f04a4a-2d5c-4d5d-8047-1e767c160fa0)

The part where I have replaced the NaN with zeros:
A. This one is during data preprocessing, where I converted the index of my dataset into time series index with hourly frequency and replace the missing value with 0.0 before converting it into a darts time series
![image](https://github.com/unit8co/darts/assets/59689052/dac3b334-cd29-45b6-af81-cb4c64a1502f)
B. I have also replaced all missing value into 0.0 when I converted my dataset using the first code snippet

Any clue or idea about the solution?

**Comment by madtoinou** [comment]
The data does not seem to contain NaN, thanks for double checking.

Since you use `n > output_chunk_length` (guessing from the size of the raw data and `output_chunk_length`), the model is doing auto-regression. It could be that some of the models predictions are NaN, can you check the coefficients of the regression model (accessing them with `model.model.coef_`)?

Maybe try to normalize the values before fitting/prediction?

**Comment by madtoinou** [comment]
The coefficients seem to be quite large, what do they look like if your fit the model using the scaled dataset?

**Comment by torivor** [comment]
The coefficients shown on the screenshot on my previous comment is the coefficient of the model that was created using the scaled dataset... Does it show any sign / clue explaining the cause of the bug? In any case, I am willing to try another model for my hierarchical forecasting use case, do you have any suggestion of another simple, fast, and light-weight model that is comparable to simple linear regression for it?? I am just trying out the basic functionalities of the darts library on my laptop before using more complex models on the company workbench

**Comment by madtoinou** [comment]
The fact that your coefficients range from 1e-5 to 1e6, while having 2018 of them (which is a lot), could potentially be the cause of the NaN (but normalized data should mitigate that). You could eventually try to fit & predict a single model for each column/component of your dataset, see if the NaN still occur.

You could eventually use XGBoost or any sklearn regression model wrapped in a darts `RegressionModel` but they might also have the NaN issue.

**Comment by madtoinou** [comment]
Hi @torivor,

Any update on this issue?
